[bug] `conan.tools.gnu.Autotools.install()` doesn't work if build machine is Windows

<!--
  Please don't forget to update the issue title.
  Include all applicable information to help us reproduce your problem.

  To help us debug your issue please explain:
-->

### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10
  * Compiler+version: MinGW 9 / gcc 11.2
  * Conan version: 1.52.0
  * Python version: 3.10.2

### Steps to reproduce (Include if Applicable)

- checkout https://github.com/conan-io/conan-center-index/pull/13038/commits/4e40e8ad67eb2d53bbe43949d7201d08b9d8651e (from https://github.com/conan-io/conan-center-index/pull/13038).
- on a windows machine call: `conan create . xz_utils/5.2.5@ -pr mingw_profile`

Fix: convert to unix_path `conanfile.package_folder` in https://github.com/conan-io/conan/blob/8aeea2230fd6375ce1da15f2380097bed9098377/conan/tools/gnu/autotools.py#L58

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
 /usr/bin/install -c -m 644  /c/users/spaceim/.conan/data/xz_utils/5.2.5/_/_/build/3cbc13750c61092b2c900ee6caef68cf1cd320b4/src/src/liblzma/api/lzma.h 'C:Usersspaceim.conandataxz_utils5.2.5__package3cbc13750c61092b2c900ee6caef68cf1cd320b4/include/.'
make[4]: Leaving directory '/c/Users/spaceim/.conan/data/xz_utils/5.2.5/_/_/build/3cbc13750c61092b2c900ee6caef68cf1cd320b4/build-release/src/liblzma/api'
make[3]: Leaving directory '/c/Users/spaceim/.conan/data/xz_utils/5.2.5/_/_/build/3cbc13750c61092b2c900ee6caef68cf1cd320b4/build-release/src/liblzma/api'
make[3]: Entering directory '/c/Users/spaceim/.conan/data/xz_utils/5.2.5/_/_/build/3cbc13750c61092b2c900ee6caef68cf1cd320b4/build-release/src/liblzma'
make[4]: Entering directory '/c/Users/spaceim/.conan/data/xz_utils/5.2.5/_/_/build/3cbc13750c61092b2c900ee6caef68cf1cd320b4/build-release/src/liblzma'
 /usr/bin/mkdir -p 'C:Usersspaceim.conandataxz_utils5.2.5__package3cbc13750c61092b2c900ee6caef68cf1cd320b4/lib'
 /usr/bin/mkdir -p 'C:Usersspaceim.conandataxz_utils5.2.5__package3cbc13750c61092b2c900ee6caef68cf1cd320b4/lib/pkgconfig'
 /bin/sh ../../libtool   --mode=install /usr/bin/install -c   liblzma.la 'C:Usersspaceim.conandataxz_utils5.2.5__package3cbc13750c61092b2c900ee6caef68cf1cd320b4/lib'
 /usr/bin/install -c -m 644 liblzma.pc 'C:Usersspaceim.conandataxz_utils5.2.5__package3cbc13750c61092b2c900ee6caef68cf1cd320b4/lib/pkgconfig'
Usage: /c/Users/spaceim/.conan/data/xz_utils/5.2.5/_/_/build/3cbc13750c61092b2c900ee6caef68cf1cd320b4/build-release/libtool [OPTION]... [MODE-ARG]...
Try 'libtool --help' for more information.
libtool:   error: 'C:Usersspaceim.conandataxz_utils5.2.5__package3cbc13750c61092b2c900ee6caef68cf1cd320b4/lib' must be an absolute directory name
make[4]: *** [Makefile:876: install-libLTLIBRARIES] Error 1
make[4]: Leaving directory '/c/Users/spaceim/.conan/data/xz_utils/5.2.5/_/_/build/3cbc13750c61092b2c900ee6caef68cf1cd320b4/build-release/src/liblzma'
make[3]: *** [Makefile:1802: install-am] Error 2
make[3]: Leaving directory '/c/Users/spaceim/.conan/data/xz_utils/5.2.5/_/_/build/3cbc13750c61092b2c900ee6caef68cf1cd320b4/build-release/src/liblzma'
make[2]: *** [Makefile:1638: install-recursive] Error 1
make[2]: Leaving directory '/c/Users/spaceim/.conan/data/xz_utils/5.2.5/_/_/build/3cbc13750c61092b2c900ee6caef68cf1cd320b4/build-release/src/liblzma'
make[1]: *** [Makefile:422: install-recursive] Error 1
make[1]: Leaving directory '/c/Users/spaceim/.conan/data/xz_utils/5.2.5/_/_/build/3cbc13750c61092b2c900ee6caef68cf1cd320b4/build-release/src'
make: *** [Makefile:616: install-recursive] Error 1
```

--- a/conan/tools/gnu/autotools.py
+++ b/conan/tools/gnu/autotools.py
@@ -6,6 +6,7 @@ from conans.client.subsystems import subsystem_path, deduce_subsystem
 from conans.client.build import join_arguments
 from conans.tools import args_to_string
 from conan.tools.files import chdir
+from conan.tools.microsoft import unix_path
 from conans.util.runners import check_output_runner
 
 
@@ -55,7 +56,7 @@ class Autotools(object):
         self._conanfile.run(command)
 
     def install(self, args=None):
-        args = args if args is not None else ["DESTDIR={}".format(self._conanfile.package_folder)]
+        args = args if args is not None else ["DESTDIR={}".format(unix_path(self._conanfile, self._conanfile.package_folder))]
         self.make(target="install", args=args)
 
     def autoreconf(self, args=None):

--- a/conans/test/functional/toolchains/gnu/autotools/test_win_bash.py
+++ b/conans/test/functional/toolchains/gnu/autotools/test_win_bash.py
@@ -47,9 +47,6 @@ def test_autotools_bash_complete():
                 autotools = Autotools(self)
                 autotools.configure()
                 autotools.make()
-
-            def package(self):
-                autotools = Autotools(self)
                 autotools.install()
         """)
 

--- a/conans/test/functional/toolchains/gnu/autotools/test_win_bash.py
+++ b/conans/test/functional/toolchains/gnu/autotools/test_win_bash.py
@@ -47,6 +47,10 @@ def test_autotools_bash_complete():
                 autotools = Autotools(self)
                 autotools.configure()
                 autotools.make()
+
+            def package(self):
+                autotools = Autotools(self)
+                autotools.install()
         """)
 
     client.save({"conanfile.py": conanfile,

